pkg/codegen/python/gen.go

```diff
@@ -2114,6 +2145,15 @@ func genPackageMetadata(
 	if pkg.Version != nil && ok && info.RespectSchemaVersion {
 		version = "\"" + PypiVersion(*pkg.Version) + "\""
 	}
+	if ok && info.InputTypes == InputTypesSettingClassesAndDicts {
+		// Add typing-extensions to the requires
+		updatedRequires := make(map[string]string, len(requires))
+		for key, value := range requires {
+			updatedRequires[key] = value
+		}
+		updatedRequires["typing-extensions"] = ">=4.11; python_version < \"3.11\""
```

Comment:
```suggestion
		updatedRequires["typing-extensions"] = ">=4.11,<5; python_version < \"3.11\""
```

I think that would be safer in case there's a new major version release with breaking changes.